### PR TITLE
HPCC-25489 LDAP credentials should be configurable as secrets

### PR DESCRIPTION
**Pull request by RussWhitehead**
Add support to externalize LDAP and other credentials as Kubernetes secrets
and Hashicorp vault secrets. Added "authn" category to secrets and vault, and
modified ECLWatch ldap.yaml to allow specification of secret name. Added README.md
markdown to document LDAP configuration using plain text yaml, k8s secrets, or
Hashicorp vault

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [x] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by RussWhitehead** [comment]
@afishbeck  Please review

File: helm/examples/ldap/README.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+   systemCommonName: "hpcc_admin"
+   systemPassword: "6b3f606902cb9a337904efe2650eb771"
+   systemUser: "hpcc_admin"
+```
+
+### Enable LDAP authentication on the ECLWatch command line
+   Add the following "auth" and "ldapAddress" settings to your ECLWatch command line or override file, ensuring the ldapAddress is the IP of your LDAP server.
+
+```bash
+esp --application=eclwatch --auth=ldap --ldapAddress=10.200.1.2 ...
```
Comment (afishbeck) [inline helm/examples/ldap/README.md:27]: This shouldn't be necessary, you should be able to set  "auth: ldap" and "ldapAddress: 10.200.1.2" in values.yaml under the esp instance.  Also needs testing to make sure it's right.

File: helm/examples/ldap/README.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+   Create a unique key name ("ldapadminsecretkey01" in this example), and set it to the secret value ("myAdminCreds") that you created above.
+
+```bash
+   secrets:
+     authn:
+       ldapadminsecretkey01: "myAdminCreds"
+```
+
+### Reference the k8s secret key
+   In the HPCC-Platform\esp\applications\common\ldap\ldap.yaml or override, set the "ldapAdminSecretKey" to the key name created above. Note that the key name "ldapadminsecretkey" is used by the LDAP security manager to resolve the secret name, and must be spelled exactly as follows.
```
Comment (afishbeck) [inline helm/examples/ldap/README.md:55]: Again, should never override the yaml files that are part of our repository.  These values can be overriden in values.yaml.

**Review by afishbeck — CHANGES_REQUESTED** [review]
@RussWhitehead comments may be a bit repetitive.. mainly about not supporting configured passwords.  We can always add some dev shortcut for passwords later if it ever makes sense.

**Pull request closed without merging**